### DISCUSSION TITLE
Tray menu displayed wrong size after menu content changes

### DISCUSSION THREAD

**Discussion by laniakea64** [hluk/CopyQ#1988] [Q&A] [answered]
Using CopyQ on Xubuntu 22.04, and overall it's awesome!  Its flexibility makes clipboard management much nicer.

Unfortunately there is an issue with the display of the tray menu: if the content of the menu changed since last time the tray menu opened, the "display size" of the menu doesn't change in light of the new menu content.  If there are more entries than when tray menu was last opened, it's too short for the content, and has scroll arrows.  If there are fewer entries, the menu will again open with scroll arrows, scrolled partially out of view, but this time covering more vertical space than is needed to display all the menu items.

If I close and immediately re-open the tray menu, only then does it have the correct display size.  But it will usually still open partially scrolled out of view.

This happens both with standard Ubuntu repository version of CopyQ and with PPA version -
```
$ copyq version
CopyQ Clipboard Manager 6.1.0~jammy
Qt: 5.15.3
KNotifications: 5.92.0
Compiler: GCC
Arch: x86_64-little_endian-lp64
OS: Ubuntu 22.04 LTS
```
It happens even in completely default configuration of CopyQ.

I have CopyQ tray icon in the Xfce "Status Tray Plugin".  Other apps tray menus don't have this problem and are sized normally.

Is this Xfce issue or CopyQ issue?
Is there a way CopyQ can fix/workaround this?

Thanks!

**Comment by hluk** [comment]
This is a bug in the tray menu integration - it does not update menu size if menu items are added dynamically when the menu shows.

Not sure if there is a good workaround I can add to CopyQ - in some older versions, the menu was created and updated in the background whenever clipboard has changed, but I removed this because it wastes CPU cycles needlessly.

The internal menu in CopyQ should work well - this is launched with `copyq menu` command.

**Reply by laniakea64** [reply]
Thank you for the reply, indeed `copyq menu` does not have this problem!  Is there a way to make clicking the tray icon run `copyq menu` instead of displaying the tray menu?

**Reply by hluk** [reply]
Unfortunately, it is not possible open the original menu - the mouse behavior and menu are overridden by the tray/notification area.

**Reply by laniakea64** [reply]
Too bad the tray menu can't be overridden.  What about the left-click action, which currently opens the CopyQ main window?  CopyQ seems to already have custom behavior there - all other apps in my tray display their tray menu on both left or right click.

For the moment, looks like I can work around this with a Xfce panel launcher that runs `copyq --start-server menu`.  This makes two CopyQ icons, but if that's what it takes to display CopyQ's working original menu on icon click...well, I'm happy that CopyQ is versatile enough that workarounds like this are possible :smiley:

**Reply by hluk** [reply, marked as answer]
Try running command `copyq config tray_menu_open_on_left_click true` and see if **left mouse button** opens the menu. **Middle mouse button** may also open the menu unless the tray area overrides the action.

**Reply by laniakea64** [reply]
> Try running command `copyq config tray_menu_open_on_left_click true` and see if **left mouse button** opens the menu. 

Works perfectly, thank you!  And this also solves https://github.com/hluk/CopyQ/issues/1981 .  Appreciate your help hluk and thanks again for making CopyQ!
